fix(wren-ai-service): The DeepSeek response may not be a valid JSON format.

luochuan-cmss opened the pull request:
When testing WrenAI with DeepSeek, I encountered a JSON formatting exception. After installing Langfuse, I noticed that DeepSeek frequently returns responses in JSON_OBJECT format with extra characters - specifically starting with \`\`\`json and ending with \`\`\`. To accommodate this behavior, we need to implement substring extraction based on curly brace positions to ensure successful JSON parsing.
Additionally, I observed that historical issues #1354  contain a similar problem.



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Enhanced response formatting by extracting and returning only the content within curly braces in AI-generated replies.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

cyyeh commented:
@luochuan-cmss sorry, the link you provided is not found. could you correct it?

luochuan-cmss commented:
> @luochuan-cmss sorry, the link you provided is not found. could you correct it?

@cyyeh Hi. I have fixed the URL to point directly to the issue. Additionally, I rebased the commits and resolved any conflicts.

cyyeh commented:
@luochuan-cmss thanks for your contribution

how about we use this code snippet instead that we make sure "```json```" is in the response?

```python
import re

def extract_braces_content(resp: str) -> str:
    """
    Extracts JSON content enclosed in a markdown code block that starts with ```json.
    Returns the JSON string including braces, or the original string if no match is found.
    """
    match = re.search(r'```json\s*(\{.*?\})\s*```', resp, re.DOTALL)
    return match.group(1) if match else resp
```

> > @luochuan-cmss sorry, the link you provided is not found. could you correct it?
> 
> @cyyeh Hi. I have fixed the URL to point directly to the issue. Additionally, I rebased the commits and resolved any conflicts.

luochuan-cmss commented:
> @luochuan-cmss thanks for your contribution
> 
> how about we use this code snippet instead that we make sure `` "```json```" `` is in the response?
> 
> ```python
> import re
> 
> def extract_braces_content(resp: str) -> str:
>     """
>     Extracts JSON content enclosed in a markdown code block that starts with ```json.
>     Returns the JSON string including braces, or the original string if no match is found.
>     """
>     match = re.search(r'```json\s*(\{.*?\})\s*```', resp, re.DOTALL)
>     return match.group(1) if match else resp
> ```
> 
> > > @luochuan-cmss sorry, the link you provided is not found. could you correct it?
> > 
> > 
> > @cyyeh Hi. I have fixed the URL to point directly to the issue. Additionally, I rebased the commits and resolved any conflicts.

@cyyeh Thank you for your advice. I've made the changes and thoroughly verified them to ensure everything is working as expected.
